### PR TITLE
Remove references to `old_{path,io}`

### PR DESCRIPTION
**Pull request by tamird**
r? @alexcrichton

### CODE REVIEW

File: src/libstd/panicking.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ use rt::{backtrace, unwind};
 use sys::stdio::Stderr;
 use thread;
 
-// Defined in this module instead of old_io::stdio so that the unwinding
+// Defined in this module instead of `io::stdio` so that the unwinding
```
Comment (tamird) [inline src/libstd/panicking.rs]: @alexcrichton this comment looks half-missing - it was first added in 5ec36c358f74fe83332231e774ea20a21d165120. Any idea what it should say?

Comment (alexcrichton) [inline src/libstd/panicking.rs]: Oh dear! I think it's fine to just delete this comment at this point!

**Comment by alexcrichton** [comment]
Nice catches!

File: src/libstd/panicking.rs (hunk 1 above)
Comment (tamird) [inline src/libstd/panicking.rs]: done

**Comment by tamird** [comment]
@alexcrichton had an errant `#![cfg_attr(test)]`. would you re-approve please?

**Comment by tamird** [comment]
found another dead file

**Pull request merged**